pkg/credentialprovider/apis/criauth/v1alpha1/types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CRIAuthPluginConfig is the configuration containing information about
+// each CRI auth plugin. Kubelet reads this configuration from disk and enables
+// each plugin as specified by the CRIAuthPlugin type.
+type CRIAuthPluginConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Plugins is a list of CRI auth plugins that will be enabled by the kubelet
+	Plugins []CRIAuthPlugin `json:"plugins"`
+}
+
+// CRIAuthPlugin is used by the kubelet container runtime to match the
+// image property string (from the container spec) with exec-based credential provider
+// plugins that provide container registry credentials.
+type CRIAuthPlugin struct {
+	// name is the name of the credential plugin.  It must match the name of the
+	// executable as seen by the kubelet.
+	Name string `json:"name"`
```

Comment:
how is the executable resolved? absolute path? relative to plugin config file? in PATH? since kubelet will execute this (often with root permissions, presumably), are we comfortable with PATH-based resolution? how do client-go exec auth plugins resolve the given command?

Reply:
Added some more comments about this. Similar to CNI, kubelet will try to exec the plugin name if it's in the plugin binary directory (configured via `--cri-auth-plugin-bin-dir` in kubelet).